lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -450,12 +450,17 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     StringRef Dir;
     StringRef File;
     SmallString<128> DirBuf;
-    SmallString<128> FileBuf;
-    std::string RemappedFileString = DebugPrefixMap.remapPath(FileName);
-    SmallString<128> RemappedFile = StringRef(RemappedFileString);
-    llvm::sys::path::remove_dots(RemappedFile);
+    SmallString<128> FileBuf(FileName);
+    llvm::sys::path::remove_dots(FileBuf);
+    std::string RemappedFileString = DebugPrefixMap.remapPath(FileBuf);
+    FileBuf.clear();
+    StringRef RemappedFile(RemappedFileString);
     std::string CurDir = DebugPrefixMap.remapPath(Opts.DebugCompilationDir);
-    if (llvm::sys::path::is_absolute(RemappedFile)) {
+    // While Opts.DebugCompilationDir is guaranteed to be absolute,
```

Comment:
Hrm is this true? I think there's a hole in this one today (that we utilize) where if you don't pass `-working-directory` this isn't set absolutely? Not sure

Reply:
Thanks for pointing that out. I reworded the comment.

Reply:
By the way, I didn't see that PR because I wasn't added a reviewer (and you probably couldn't add me due to how the repository is set up). Feel free to @-mention in PRs if you want a review.

Reply:
IIUC, this patch also fixes #39695 or are there other edge cases that I'm missing?

Reply:
Thanks, I'm not entirely sure, but I think so, my test case is a bit different though with the nested directory case, can you add that same test case here?

Reply:
In my testcase the file is `%t/SubdirectoryWithAVeryUniqueName/debug_prefix_map_abs_rel.swift` and we remap `%t=.`.
In yours the file is `%/t/nested/dir/foo.swift` and we remap `%t/nested=.`. Since `%t` is the full path into the build directory, I don't see how the extra `nested` subdir makes a difference? Let me know if I'm missing something.

Reply:
Hrm I thought there was a difference and it had something to do with `pwd` differing, but with some minor changes around the placement of `.` and `./` this change passes the tests from my PR